editor.planx.uk/src/@planx/components/PlanningConstraints/utils.test.ts

```diff
@@ -0,0 +1,145 @@
+import { defaultIntersectingConstraints, defaultNots, mockConstraints } from "./mocks/overrides/constraints";
+import { DEFAULT_FN } from "./model";
+import { handleOverrides } from "./utils";
+
+describe("handleOverrides basic", () => {
+  test("when no constraints are reported as inaccurate, it returns the original intersections and nots", () => {
+    const overrides = undefined;
+    const result = handleOverrides(DEFAULT_FN, mockConstraints, overrides, defaultIntersectingConstraints, defaultNots);
+
+    expect(result).toEqual({
+      intersectingConstraints: defaultIntersectingConstraints,
+      nots: defaultNots,
+    });
+  });
+
+  test.skip("when a constraint was reported as inaccurate, but is not part of the latest GIS API response, it's skipped", () => {
+    const overrides = {
+      "listed": {
+        "entities": ["some-outdated-entity"],
+        "reason": "This isn't here anymore",
+      },
+    };
+    const result = handleOverrides(DEFAULT_FN, mockConstraints, overrides, defaultIntersectingConstraints, defaultNots);
+
+    expect(result).toEqual({
+      intersectingConstraints: defaultIntersectingConstraints,
+      nots: defaultNots,
+    });
+  });
+
+  test("when many constraints are reported as inaccurate, all categories are correctly updated", () => {
+    const overrides = {
+      "designated.conservationArea": {
+        "entities": ["44006848"],
+        "reason": "I'm not in a conservation area"
+      },
+      "listed": {
+        "entities": ["42101437"],
+        "reason": "I don't recognize this listed building name, I don't think it's for my property"
+      },
+    };
+    const result = handleOverrides(DEFAULT_FN, mockConstraints, overrides, defaultIntersectingConstraints, defaultNots);
+
+    expect(result["intersectingConstraints"][DEFAULT_FN]).not.toContain("designated.conservationArea");
+    expect(result["intersectingConstraints"][DEFAULT_FN]).not.toContain("designated");
+    expect(result["intersectingConstraints"][DEFAULT_FN]).toContain("listed");
+    expect(result["intersectingConstraints"][DEFAULT_FN]).toContain("listed.grade.II");
+  });
+});
+
+describe("handleOverrides with non-granular constraint categories", () => {
+  test("when all entities in a constraint category are inaccurate, the whole category no longer applies", () => {
+    const overrides = {
+      "road.classified": {
+        "entities": ['undefined'], // @todo roads don't have entity ids, record the property.OBJECTID via OS?
```

Comment:
This test case highlights how `road.classified` is a still a bit of a strange edge case:
- Because it's returned via OS and not Planning Data, it doesn't match the "entity" type we compare throughout this function (see the `mockConstraints` data as an example)
- But because of our UPRN matching method, we'll only ever return ONE applicable classified road, so the `handleOverride` always fits the first scenario of "every entity is inaccurate, therefore whole category is" and still works as expected :heavy_check_mark: 
- What I need to come back and fix is the `_overrides` audit record, which is currently incorrectly recording 'undefined' !